config/ingest.go

```diff
@@ -14,4 +15,6 @@ type Ingest struct {
 	// StoreBatchSize is the number of entries in each write to the value
 	// store.  Specifying a value less than 2 disables batching.
 	StoreBatchSize int
+	// SelectorRecursionLimitDepth sets the maximum recursion depth for the selector used to select advertisements.
+	SelectorRecursionLimitDepth int
```

Comment:
How do I set this to unlimited?  Specify `0` or `-1`?

Reply:
> Also, maybe there should be something similar to this:
> https://github.com/filecoin-project/index-provider/blob/main/config/config.go#L97-L98

Off topic: I am curious why call this `defaults`? would something like `reset` or even a constructor for the config struct type like `newIngestConfig` that constructs the type with default values set make sense?

Reply:
> How do I set this to unlimited? Specify 0 or -1?

It looks like this depends on `RecursionLimit` mode: There is a mode dedicated to `RecursionLimit_None` i.e. no limits to recursion. It is unclear to me how to set the limit on _depth_ to unlimited.

I'll add some gluecode here to switch the mode to `None` if `SelectorRecursionLimitDepth` is set to `-1`.

Reply:
> why call this defaults? would something like ... `newIngestConfig` that constructs the type with default values set make sense?

The `defaults` function is run _after_ the config is loaded, to replace any zero-valued items with the appropriate real default value (when it is different from an item's zero-value).  So, a constructor would not be quite right because the config object is already built/loaded.

If `SelectorRecursionLimitDepth` is not configured in the config file, I would expect a default value to be used.  Similarly, if I set it to the zero-value (0 in this case), I would also expect some reasonable default to be used.

Reply:
minor:  It sounds better to me to call this `SelectorRecursionDepthLimit`.  "LimitDepth" sounds more like a boolean deciding whether to limit the depth, whereas "DepthLimit" sounds like a maximum depth.

Reply:
Note that I have removed this config completely from this PR, and captured #123 which depends on go-legs changes to expose limit.